### PR TITLE
Add define plugin in allowed plugins

### PR DESCRIPTION
**Pull request by fapspirit**
Hello!
I added `define` plugin in the `allowedPlugins` and here is why:
In `vue.config.js` I added `definePlugin` to pass some env variables (hosts, etc.), but since this plugin isn't allowed, my env variables became undefined. This can be solved by adding plugin in custom webpack config, specified in `config/storybook/webpack.config.js`, but keeping the same logic in two different places seems to be wrong.

### CODE REVIEW

**Comment by fapspirit** [comment]
By the way, what do you think about possibility to extend allowedPlugins manually by user? This is possible through custom plugin property in `vue.config.js`

**Comment by pksunkara** [comment]
I think that is probably the better way to do this.

**Comment by pksunkara** [comment]
Can you give me a sample repo where you are encountering the issue you are trying to fix?

**Comment by fapspirit** [comment]
Sure, will do it later this day.
Also, i can help with implementing this feature!

**Comment by fapspirit** [comment]
@pksunkara There is repo with example and steps to reproduce the issue
https://github.com/fapspirit/storybook_example

**Comment by pksunkara** [comment]
More information, https://github.com/storybooks/storybook/issues/4171

**Comment by fapspirit** [comment]
Oh, I see that the problem described in my example can be solved with `.env` files. However, my work use-cases can't be solved with `dotenv` package mostly because we need to pass JSON that is generated during the runtime, so we created our own solution for this purpose.
If there is no reason due to which we can't include `define` in the allowed plugins without breaking anything, can we return it back? Otherwise, we can still implement extending as a property from `vue.config.js`

**Comment by pksunkara** [comment]
I would like to do it as an option for storybook plugin.

> we can still implement extending as a property from vue.config.js

I think you mean the same, right? (I don't recall the term `property` in vue-cli). I have also added CI and chromatic. So, please rebase your work and let's start testing this plugin.

**Comment by fapspirit** [comment]
@pksunkara I added support for extending allowed plugins through plugin options in `vue.config.js`. This solution extends default allowed plugins only with unique values passed by user

File: lib/wrapInitialConfig.js [hunk 1]
```diff
@@ -10,6 +10,15 @@ module.exports = (api) => {
     'hash-module-ids',
   ];
 
+  if (options && options.allowedPlugins) {
```
Comment (pksunkara) [inline lib/wrapInitialConfig.js]: This block looks too complicated. I don't think you need this at all.

File: lib/wrapInitialConfig.js [hunk 2]
```diff
@@ -10,6 +10,15 @@ module.exports = (api) => {
...
+    allowedPlugins = options.allowedPlugins.reduce((allPlugins, plugin) => {
+      if (allPlugins.indexOf(plugin) < 0) {
+        allPlugins.push(plugin);
+      }
+      return allPlugins;
+    }, allowedPlugins);
+  }
+
   existingPlugins.forEach((plugin) => {
     if (!allowedPlugins.includes(plugin)) {
```
Comment (pksunkara) [inline lib/wrapInitialConfig.js]: You can do something like `if (!allowedPlugins.includes(plugin) && !options.allowedPlugins.includes(plugin)) {`

**Comment by pksunkara** [comment]
I don't know why the tests are not running.

**Comment by fapspirit** [comment]
Maybe tests are not running because of stale pull-request? I can create new one

**Comment by pksunkara** [comment]
Please do then. Thanks. I will add a test for this later on.

File: lib/wrapInitialConfig.js (hunk 1 above)
Comment (pksunkara) [reply lib/wrapInitialConfig.js]: Yeah something like `options.allowedPlugins ||= []`

Comment (fapspirit) [reply lib/wrapInitialConfig.js]: What about to describe default options object and merge it with `Object.assign` or with `webpack-merge`, which is dependency for `@vue/cli-service`?

Comment (pksunkara) [reply lib/wrapInitialConfig.js]: We can do that in index.js. Let's use `Object.assign`

**Comment by pksunkara** [comment]
Can you also add a little bit of documentation?

File: lib/wrapInitialConfig.js (hunk 2 above)
Comment (fapspirit) [reply lib/wrapInitialConfig.js]: Just wanted to extend existing array in case we need to use it in future. if this is not necessary, I can do as you say

**Comment by fapspirit** [comment]
Sure! Will do a new branch for new pull request after we finish discussion about implementation

**Comment by fapspirit** [comment]
Moved here #32

**Pull request closed without merging**